### PR TITLE
Fixes the CardView customization mentioned in #1582

### PR DESCRIPTION
**Pull request by kunzef**
## Issue

This fixes the following issue(s):
- Resolves: #1582 

## What it does
This PR rewrites the way Cards handle its discard and alyways hide functionality. It moves all information stored for the card in regards to the discard functionality into one SharedPreference per Card. This shared preference can then be completly wiped when restoring the card. The alyways hide functionality is set in the default SharedPreferences by a key defined in the new CardType enum. This key is stored in the String resource file making it easy to use across the Application and preventing accidental differnces in these keys. The usage of String resource also allows for seamless integration into the Settings.xml ensuring synchronity between the state of the Card. Also settings option for every Card were added because previously there was no way to restore these once alyawys hide was pressed. The CardTypes enum and the combination with the string resource also makes the addition of new Cards less error prone as it is more centralized. The PR also fixes a bug when refreshing the CardView leading to an error when using doAsync from Anko and a crash when using coroutines as mentioned [here](https://github.com/TUM-Dev/Campus-Android/issues/1581#issuecomment-1571176965).

## Why this is useful for all students
This PR fixes the issues mentioned in #1582. Thus enhancing the user experience by restoring the excepted functionality.

### CODE REVIEW

File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/overview/MainFragment.kt [hunk 1]
```diff
@@ -126,6 +126,7 @@ class MainFragment : BaseFragment<Unit>(
         swipeRefreshLayout.isRefreshing = true
         onRefresh()
         downloadNewsAlert()
+        println("Refresh finished")
```
Comment (tobiasjungmann) [inline app/src/main/java/de/tum/in/tumcampusapp/component/ui/overview/MainFragment.kt:129]: Ideally use logging or remove it to make it more coherent with the rest

**Review by tobiasjungmann — CHANGES_REQUESTED** [review]
Please try to not format the whole file in the future - either use a different PR for reformatting / adapt the formatting rules in your IDE to the previously used version. That also happens to me every time :D
Some of your commit messages can be more intuitive ;)
Add some sort of comment (e.g. "Value is only written in the editor, changes are not yet copied back in the shared preferences.") to line 118 in `card.kt` to make obvious that the changes are not yet applied. I cannot annotate the function with a comment myself since there are no changes. Afterwards it should be ready to merge!

**Comment by kunzef** [comment]
> Please try to not format the whole file in the future - either use a different PR for reformatting / adapt the formatting rules in your IDE to the previously used version. 

Are you refering to the first `Fixed formatting` commit? I think my error there was to accept the option _Remove custom line breaks_.
Otherwise I adopte the suggested changes.
Thanks for the feedback.

**Comment by kunzef** [comment]
I was able to fix both issues. However, the fix for the visibility of the news card does not fix the root cause described here #1597. But it will work perfectly for the startpage without further issues.

**Review by tobiasjungmann — APPROVED** [review]
Amazing, thank you for looking into it! I could not spot any bug anymore.

**Pull request merged**